### PR TITLE
editor: Add automatic markdown list continuation on newline and indent on tab

### LINKED ISSUE #5089: Detect lists and automatically insert the appropriate item
### Check for existing issues

- [X] Completed

### Is your feature request related to a problem?

- ordered list: `1. `
- unordered list: `- `
- task: `- [ ] `

### Describe the solution you'd like

Zed 0.66.0 – /Applications/Zed Preview.app
macOS 13.0.1
architecture arm64

### If applicable, add mockups / screenshots to help present your vision of the feature

_No response_

### PR DESCRIPTION
**Pull request by 0xRaduan**
Closes #5089

Release notes:
- Markdown lists now continue automatically when you press Enter (unordered, ordered, and task lists). This can be configured with `extend_list_on_newline` (default: true).
- You can now indent list markers with Tab to quickly create nested lists. This can be configured with `indent_list_on_tab` (default: true).

### CODE REVIEW

**Comment by 0xRaduan** [comment]
@cla-bot check

**Comment by 0xRaduan** [comment]
@cla-bot check

**Comment by 0xRaduan** [comment]
@cla-bot check

File: assets/settings/default.json [hunk 1]
```diff
@@ -1131,6 +1131,8 @@
   "remove_trailing_whitespace_on_save": true,
   // Whether to start a new line with a comment when a previous line is a comment as well.
   "extend_comment_on_newline": true,
+  // Whether to continue markdown lists when pressing enter.
+  "extend_list_on_newline": true,
```
Comment (0xRaduan) [inline assets/settings/default.json:1182]: can potentially be:
```suggestion
  "extend_markdown_list_on_newline": true,
```

File: crates/editor/src/editor_tests.rs [hunk 2]
```diff
@@ -27988,3 +27988,154 @@ async fn test_multibuffer_selections_with_folding(cx: &mut TestAppContext) {
...
+            Some(Arc::new(Language::new(
+                LanguageConfig {
+                    name: "Markdown".into(),
+                    ..LanguageConfig::default()
+                },
+                None,
+            ))),
+            cx,
+        )
+    });
```
Comment (0xRaduan) [inline crates/editor/src/editor_tests.rs]: probably should add a check for non markdown language with `extend_list_on_newline` = true; right now is missing

**Review by 0xRaduan — COMMENTED** [review]
tiny self-review

**Comment by Feel-ix-343** [comment]
very excited for this

**Comment by 0xRaduan** [comment]
fixed tests + rebaesd on latest master

**Review by smitbarmase — APPROVED** [review]
This is a great improvement. I made a bunch of fixes to handle edge cases, and added a few things:

- Instead of hardcoded logic, it’s now configurable per language and supports Markdown out of the box.
- Added a way to indent list items with `Tab`, without that this feature felt incomplete.
- Did some refactoring and stylistic cleanup around how we handle these prefix scenarios overall, including comments, doc comments, and the new configs.
- Added settings to configure `Tab` and `newline` behavior separately.

Thank you so much for your work!
